lib/cisco_node_utils/radius_server.rb

```diff
@@ -0,0 +1,258 @@
+# Radius Server provider class
+
+# Jonathan Tripathy et al., September 2015
+
+# Copyright (c) 2014-2015 Cisco and/or its affiliates.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require File.join(File.dirname(__FILE__), 'node')
+
+module Cisco
+  # RadiusServer - node utility class for
+  # Raidus Server configuration management
+  class RadiusServer < NodeUtil
+    attr_reader :name
+
+    def initialize(name, instantiate=true)
+      unless name[/^(?:[0-9]{1,3}\.){3}[0-9]{1,3}$/]
+        fail ArgumentError, 'Invalid value(IP is not an IP address)'
+      end
+      @name = name
+
+      create if instantiate
+    end
+
+    def self.radiusservers
+      radiusservers_list = config_get('radius_server', 'hosts')
+      return hash if radiusservers_list.nil?
+
+      radiusservers_list.each do |id|
+        hash[id] = RadiusServer.new(id)
+      end
+
+      hash
+    end
+
+    def create
+      config_set('radius_server',
+                 'hosts',
+                 state: '',
+                 ip:    @name)
+    end
+
+    def destroy
+      config_set('radius_server',
+                 'hosts',
+                 state: 'no',
+                 ip:    @name)
+    end
+
+    def ==(other)
+      name == other.name
+    end
+
+    def auth_port
+      val = config_get('radius_server', 'auth-port', @name)
+      val = val[0].to_i unless val.nil?
+      val
+    end
+
+    def auth_port=(val)
+      unless val.nil?
```

Comment:
Our convention in most of the other classes is to let `''` represent 'not configured at all' rather than `nil`. For example, interface.rb:

    def vrf
      vrf = config_get('interface', 'vrf', @name)
      return '' if vrf.nil?
      vrf.shift.strip
    end

    def vrf=(vrf)
      fail TypeError unless vrf.is_a?(String)
      if vrf.empty?
        config_set('interface', 'vrf', @name, 'no', '')
      else
        config_set('interface', 'vrf', @name, '', vrf)
      end

However, if a property has a default value (it is never truly 'removed'), then we should do that instead:

    def access_vlan
      vlan = config_get('interface', 'access_vlan', @name)
      return default_access_vlan if vlan.nil?
      vlan.shift.to_i
    end

    def access_vlan=(vlan)
      config_set('interface', 'access_vlan', @name, vlan)